### DISCUSSION TITLE
Schedule for rolling out releases to hosted

### DISCUSSION THREAD

**Discussion by Cerno-b** [WeblateOrg/weblate#10223] [Q&A] [answered]
Hi!

I wasn't able to find which release of Weblate is currently running on https://hosted.weblate.org/, can I assume that it gets updated to the most recent version automatically? Is this a global update or does it work on a per-project basis?

Since there was a new release just yesterday, is there some way for me to track when the update happens on https://hosted.weblate.org/? Or is it already live?

**Comment by m-aciek** [comment]
You can check it at https://hosted.weblate.org/about/, in the third section. The version numbers are displayed on desktop only, not on mobile.

**Comment by Cerno-b** [comment]
I don't see it. Do I need special access?

On the about tab there are three sections, `About Weblate`, `Hosting translations` and `Weblate is built on libre software`.

Then there are three tabs at the top `About Weblate`, `Statistics` and `Keys`

**Reply by nijel** [reply]
Just to clarify this, the version on this page is hidden since https://github.com/WeblateOrg/weblate/issues/10160

**Comment by nijel** [comment]
There is no schedule, it is deployed manually when there is reason to do so. Version info is currently hidden because some users got scared of the -dev version.

**Reply by Geeyun-JY3** [reply]
TBH, I still hope that the main version info of Weblate can be unhidden for Hosted Weblate. 
Currently (after https://github.com/WeblateOrg/weblate/commit/dd716f9a6af53dc3c5c848f67f82229ca43e4d03 was deployed), there is no way for the normal user to know the main version number of the Weblate that Hosted Weblate is using. Considering that Hosted Weblate always uses the latest main version with the recent commits, I think hiding the commit hash of the Weblate version info is enough to avoid the potential security risk.
Just show the main version number like `Weblate 5.0-dev`.

**Reply by nijel** [reply]
I don't see a security risk in showing versions.  Especially in this case, where it is nearly always the latest version of all packages. 

Showing `-dev` or frequent version changes were scaring some users (for example, when the major version changed from 4 to 5 people expected breaking changes). That's why we hide it now.

On the other side, this feature was created for corporations that have a security policy of hiding version information,  so it hides it completely. And I don't want to introduce yet another level of configuring this just for Hosted Weblate.

Also, I fail to see use case here. When somebody reports issue with our service,  we know what code it runs to be able to diagnose it.

**Reply by m-aciek** [reply]
In my case it was waiting for deployment of new features. I used to check the about page on Hosted Weblate quite frequently to know if I can start using the categories I was looking forward to. For me, a notification about a deployment of a new feature version of Weblate (with a link to release notes?) would be a very helpful thing.

**Reply by nijel** [reply]
As the deployments are quite frequent, I don't think it's reasonable to announce each of that somehow.

**Comment by Cerno-b** [comment]
@nijel Could you give a rough ballpark estimate for how long it usually takes for a release to go live on hosted? "when there is reason to do so" could be within a week, but it could also be 2 years. ;)

Would you be willing to share what the current version is on hosted and when you plan to integrate the newest release?

@Geeyun-JY3 I understand that hiding the version can be seen as a security measure, and I would understand if it's not public. For self-hosted websites I guess it makes sense to hide the version because exploiters could just scrape the web for vulnerable instances and attack them automatically.

For the official hosted weblate this might be less of an issue, but if it's still a concern, would it be okay to publish the current version somewhere else where it's less obvious to look up? Maybe by editing the corresponding release note to reflect which of the releases is the current hosted version?

Ultimately, I agree with @Geeyun-JY3, just suggesting alternatives if this is not possible.

**Reply by nijel** [reply, marked as answer]
Typically, it takes hours to days to deploy a new version. Mostly, the deployment is postponed for technical reasons (like when the database migration is expected to take hours and cannot be performed outside a proper maintenance window).

**Reply by Cerno-b** [reply]
That's good to know. If it's this fast I don't think I need a version on hosted. 

Just one final point of clarification: When you say deploy a new version, is the based on the releases or on the head of the main branch? So do you update the hosted whenever there are new commits or whenever there are new releases?

**Reply by nijel** [reply]
We typically deploy from the `main` branch.
